src/main/java/com/yahoo/bullet/operations/aggregations/TopK.java

```diff
@@ -0,0 +1,112 @@
+
+    private final String newName;
+
+    /**
+     * Constructor that requires an {@link Aggregation}.
+     *
+     * @param aggregation An {@link Aggregation} with valid fields and attributes for this aggregation type.
+     */
+    @SuppressWarnings("unchecked")
+    public TopK(Aggregation aggregation) {
+        super(aggregation);
+
+        String errorConfiguration = (config.getOrDefault(BulletConfig.TOP_K_AGGREGATION_SKETCH_ERROR_TYPE,
+                                                         DEFAULT_ERROR_TYPE)).toString();
+
+        ErrorType errorType = getErrorType(errorConfiguration);
+
+        Map<String, Object> attributes = aggregation.getAttributes();
+
+        newName = attributes == null ? DEFAULT_NEW_NAME :
+                                       attributes.getOrDefault(NEW_NAME_FIELD, DEFAULT_NEW_NAME).toString();
+
+        int maxMapSize = getMaxMapEntries(config);
+        Number threshold = getThreshold(attributes);
+        int size = aggregation.getSize();
+        sketch = threshold != null ? new FrequentItemsSketch(errorType, maxMapSize, threshold.longValue(), size) :
+                                     new FrequentItemsSketch(errorType, maxMapSize, size);
+    }
+
+    @Override
+    public List<Error> initialize() {
+        if (Utilities.isEmpty(fields)) {
+            return singletonList(REQUIRES_FIELD_ERROR);
+        }
+        return null;
+    }
+
+    @Override
+    public void consume(BulletRecord data) {
+        sketch.update(composeField(data));
+    }
+
+    @Override
+    public Clip getAggregation() {
+        Clip result = super.getAggregation();
+        result.getRecords().forEach(this::splitFields);
+        return result;
+    }
+
+    private void splitFields(BulletRecord record) {
+        String field = record.getAndRemove(FrequentItemsSketch.ITEM_FIELD).toString();
+        List<String> values = decomposeField(field);
+        for (int i = 0; i < fields.size(); ++i) {
+            String originalField = fields.get(i);
+            String fieldName = fieldsToNames.get(originalField);
+            record.setString(Utilities.isEmpty(fieldName) ? originalField : fieldName, values.get(i));
+        }
+        record.rename(FrequentItemsSketch.COUNT_FIELD, newName);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static int getMaxMapEntries(Map config) {
+        int entries = ((Number) config.getOrDefault(BulletConfig.TOP_K_AGGREGATION_SKETCH_ENTRIES,
+                                                   DEFAULT_MAX_MAP_ENTRIES)).intValue();
+        // If non-positive or not a power of 2
+        if (entries <= 0 || (entries & entries - 1) != 0) {
+            return DEFAULT_MAX_MAP_ENTRIES;
+        }
+        return entries;
+    }
+
+    private static Number getThreshold(Map<String, Object> attributes)  {
+        if (Utilities.isEmpty(attributes)) {
+            return null;
+        }
+        return Utilities.getCasted(attributes, THRESHOLD_FIELD, Number.class);
```

Comment:
Cool, maybe we can add a "warning"s that we can return in the metaData in the future or something...